dlt/common/configuration/inject.py

```diff
@@ -88,15 +97,16 @@ def decorator(f: TFun) -> TFun:
         )
         spec_arg: Parameter = None
         pipeline_name_arg: Parameter = None
-
         if spec is None:
-            SPEC = spec_from_signature(f, sig, include_defaults)
+            SPEC = spec_from_signature(f, sig, include_defaults, base=base)
         else:
             SPEC = spec
 
         if SPEC is None:
             return f
 
+        func_id = id(f)
```

Comment:
we use the id of the wrapped function as part of the key to created the threadlock, if the same function is wrapped multiple times, then the key will collide, so we could also consider a uniqid